bot/src/authorisation.ts

```diff
@@ -0,0 +1,73 @@
+import getSigner from 'auctions-core/src/signer';
+import {
+    authorizeCollateral,
+    authorizeWallet,
+    getCollateralAuthorizationStatus,
+    getWalletAuthorizationStatus,
+} from 'auctions-core/src/authorizations';
+import { ETHEREUM_NETWORK, KEEPER_PREAUTHORIZE, WHITELISTED_COLLATERALS } from './variables';
+import { parseCollateralWhitelist } from './whitelist';
+
+export async function authorizeKeeperWallet(authorizeCallback?: () => void) {
+    const signer = await getSigner(ETHEREUM_NETWORK);
+    const walletAddress = await signer.getAddress();
+    const isWalletAuth = await getWalletAuthorizationStatus(ETHEREUM_NETWORK, walletAddress);
+
+    if (!isWalletAuth) {
+        console.info(`keeper: wallet "${walletAddress}" has not been authorized yet. Attempting authorization now...`);
+        const transactionHash = await authorizeWallet(ETHEREUM_NETWORK, walletAddress, false);
+        console.info(`keeper: wallet "${walletAddress}" successfully authorized via "${transactionHash}" transaction`);
+        if (authorizeCallback) {
+            authorizeCallback();
+        }
+    }
+}
+
+export async function checkAndAuthorizeKeeperCollateral(
+    walletAddress: string,
+    collateralType: string,
+    authorizeCallback?: () => void
+) {
+    // get collateral authorization status
+    const isCollateralAuth = await getCollateralAuthorizationStatus(ETHEREUM_NETWORK, collateralType, walletAddress);
+
+    // try to authorize the collateral then return
+    if (!isCollateralAuth) {
+        console.info(
+            `keeper: collateral "${collateralType}" has not been authorized on wallet "${walletAddress}" yet. Attempting authorization now...`
+        );
+        const collateralTransactionHash = await authorizeCollateral(
+            ETHEREUM_NETWORK,
+            walletAddress,
+            collateralType,
+            false
+        );
+        console.info(
+            `keeper: collateral "${collateralType}" successfully authorized on wallet "${walletAddress}" via "${collateralTransactionHash}" transaction`
+        );
+        if (authorizeCallback) {
+            authorizeCallback();
+        }
```

Comment:
Same here:
- callback is unnecessary 
- else should print "... already authorized ..."

Reply:
See above.